src/Param.h

```diff
@@ -120,13 +120,19 @@ class Param {
     }
 
     void set_min_value(Expr min) {
+        if (!min.defined()) {
```

Comment:
I'm able to trigger the crash in either AOT or JIT mode with just this line:

`Halide::Param<float> p{"p", 1.0f, 0.0f, Halide::Expr()};`